src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,256 @@
+            {
+                Trace.Error($"The autologon user '{logonAccount}' doesnt have a user profile on the machine. Please login once with the expected autologon user and reconfigure the agent again");
+                throw new InvalidOperationException(StringUtil.Loc("NoUserProfile", logonAccount));
+            }
+
+            DisplayWarningsIfAny(regHelper);        
+            UpdateRegistriesForAutoLogon(regHelper, userName, domainName, logonPassword);
+            ConfigurePowerOptions();
+        }
+
+        public bool RestartNeeded()
+        {
+            var regHelper = new AutoLogonRegistryManager(HostContext.GetService<IWindowsRegistryManager>());            
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if (string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException(StringUtil.Loc("AutoLogonNotConfigured"));
+            }
+
+            return !_windowsServiceHelper.HasActiveSession(domainName, userName);
+        }
+
+        public void Unconfigure()
+        {
+            if(!_windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                Trace.Error("Needs Administrator privileges to unconfigure an agent running with AutoLogon capability.");          
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonRemoval"));
+            }
+            
+            /* We need to find out first if the AutoLogon was configured for the same user
+            If it is different user we should be reverting the AutoLogon user specific registries
+            */
+            var regManager = HostContext.GetService<IWindowsRegistryManager>();
+            var regHelper = new AutoLogonRegistryManager(regManager);
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if (string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException(StringUtil.Loc("AutoLogonNotConfigured"));
+            }
+
+            if (_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                Trace.Info($"AutoLogon is enabled for the same user, reverting the registry settings now.");
+                regHelper.RevertOriginalRegistrySettings();
+            }
+            else
+            {
+                Trace.Info($"AutoLogon is enabled for the different user {domainName}\\{userName}, reverting the registry settings now.");
+                var securityIdForTheUser = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                AutoLogonRegistryManager regHelperForDiffUser = new AutoLogonRegistryManager(regManager, securityIdForTheUser);
+                regHelperForDiffUser.RevertOriginalRegistrySettings();
+            }            
+        }
+
+        public bool IsAutoLogonConfigured()
```

Comment:
OK I was confused. I was conflating auto-logon registry settings (hklm) and startup-process registry settings (hkcu). Here is what I think makes sense:
1. Split the AutoLogonRegistryManager into two classes: AutoLogonRegistryManager and StartProcessRegistryManager (or if some other name makes sense).
2. Write a file `.autologon` (same pattern as other . files). And put some indication in that file who the user is. Regardless if this is redundant (already in hklm), at a minimum this is helpful informationally. Also this help during unconfigure to identify the target user.
3. During unconfig, change use IConfigurationStore.GetAutoLogonUserName() to determine if auto-logon is configured and to get the user name.